weave-js/src/core/ops/domain/artifactVersion.ts

```diff
@@ -114,6 +115,60 @@ export const opArtifactVersionSize = makeArtifactVersionOp({
   resolver: ({artifactVersion}) => artifactVersion.size,
 });
 
+export const opArtifactVersionTTLDurationSeconds = makeArtifactVersionOp({
+  name: 'artifactVersion-ttlDurationSeconds',
+  argTypes: artifactVersionArgTypes,
+  description: `Returns the ttl duration in seconds that ${docType(
+    'artifactVersion'
+  )} takes to mark as deleted`,
+  argDescriptions: {
+    artifactVersion: artifactVersionArgDescription,
+  },
+  returnValueDescription: `The ttl duration in seconds for ${docType(
+    'artifactVersion'
+  )}`,
+  returnType: inputTypes => 'number',
+  resolver: ({artifactVersion}) => artifactVersion.ttlDurationSeconds,
+});
+
+export const opArtifactVersionTTLDaysLeft = makeArtifactVersionOp({
+  name: 'artifactVersion-ttlDaysLeft',
+  argTypes: artifactVersionArgTypes,
+  description: `Returns the time to live(ttl) days left that ${docType(
+    'artifactVersion'
+  )} has before its deleted`,
+  argDescriptions: {
+    artifactVersion: artifactVersionArgDescription,
+  },
+  returnValueDescription: `The ttl days left in string form for UI for ${docType(
+    'artifactVersion'
+  )}`,
+  returnType: inputTypes => 'string',
+  resolver: ({artifactVersion}) => {
+    if (artifactVersion.ttlDurationSeconds != null) {
```

Comment:
Update after https://github.com/wandb/core/pull/16442
```suggestion
    if (artifactVersion.ttlDurationSeconds != -2) {
```